### PR TITLE
{cae}[foss/2021a] OpenFOAM v9

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

requires https://github.com/easybuilders/easybuild-easyblocks/pull/2544 + ~#13266~ (CGAL)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3502.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/31929d96e77af8e3ae52e5cc59fecf7a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3170.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/05ab618e49a9c53b9baac28592639f5f for a full test report.

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ OpenFOAM has an extensive range of features to solve anything from complex fluid flows
+ involving chemical reactions, turbulence and heat transfer,
+ to solid dynamics and electromagnetics."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+# toolchainopts = {'cstd': 'c++11'}
+
+source_urls = ['https://github.com/OpenFOAM/OpenFOAM-%(version_major)s/archive']
+sources = ['version-%(version)s.tar.gz']
+patches = ['OpenFOAM-%(version)s-cleanup.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb:15]: I forgot to include the patch file, yet the test suite still passed?!

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+version = '9'
+
+homepage = 'https://www.openfoam.org/'
+description = """OpenFOAM is a free, open source CFD software package.
+ OpenFOAM has an extensive range of features to solve anything from complex fluid flows
+ involving chemical reactions, turbulence and heat transfer,
+ to solid dynamics and electromagnetics."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+# toolchainopts = {'cstd': 'c++11'}
```
Comment (branfosj) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb:11]:
```suggestion
```

If this is not required then we can remove it.

**Comment by branfosj** [comment]
Also requires https://github.com/easybuilders/easybuild-easyblocks/pull/2544 ?

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb:15]: The test suite failed, but we did not get the failure - from https://github.com/easybuilders/easybuild-easyconfigs/runs/3363235309#step:8:131:
```
2021-08-18T16:22:50.8554539Z ======================================================================
2021-08-18T16:22:50.8557882Z FAIL: test__parse_easyconfig_OpenFOAM-9-foss-2021a.eb (test.easyconfigs.easyconfigs.EasyConfigTest)
2021-08-18T16:22:50.8560934Z Test for easyconfig OpenFOAM-9-foss-2021a.eb
2021-08-18T16:22:50.8562783Z ----------------------------------------------------------------------
2021-08-18T16:22:50.8563401Z Traceback (most recent call last):
2021-08-18T16:22:50.8565235Z   File "test/easyconfigs/easyconfigs.py", line 1348, in innertest
2021-08-18T16:22:50.8566038Z     template_easyconfig_test(self, spec_path)
2021-08-18T16:22:50.8566947Z   File "test/easyconfigs/easyconfigs.py", line 1223, in template_easyconfig_test
2021-08-18T16:22:50.8567826Z     self.assertTrue(os.path.isfile(patch_full), msg)
2021-08-18T16:22:50.8569643Z AssertionError: Patch file /home/runner/work/easybuild-easyconfigs/easybuild-easyconfigs/easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-cleanup.patch is available for OpenFOAM-9-foss-2021a.eb
2021-08-18T16:22:50.8570775Z 
2021-08-18T16:22:50.8571354Z ----------------------------------------------------------------------
2021-08-18T16:22:50.8571960Z Ran 13091 tests in 505.157s
2021-08-18T16:22:50.8572208Z 
2021-08-18T16:22:50.8572524Z FAILED (failures=1)
```

Comment (branfosj) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-9-foss-2021a.eb:15]: Broken by #13745 - I've add a revert PR in #13770

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3170.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/458ca2c44489bc39e8e76165aa7613bf for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3502.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/259d4f1af9cd181d9289a64a48a0a591 for a full test report.

**Comment by boegel** [comment]
> Also requires [easybuilders/easybuild-easyblocks#2544](https://github.com/easybuilders/easybuild-easyblocks/pull/2544) ?

Yes, of course, overlooked that... 🤦‍♂️

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16
EB_ARGS="--include-easyblocks-from-pr 2544"

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2544
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3585.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/c4029372ed68b43dce44a190c0b6e600 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2544
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/d0919798eff5bc24d41cc14e95639c45 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2544
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3111.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/800dcbc6d07804f31156f0cc4d45854c for a full test report.

**Review by branfosj — APPROVED** [review]
lgtm

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**